### PR TITLE
fix(react): Fix problem with React not showing underlying SFC's name

### PR DESCRIPTION
**Pull request by shaine**

### CODE REVIEW

File: src/error-catch-notifier.js [hunk 1]
```diff
@@ -219,7 +219,14 @@ export function wrap(targetFunction, options) {
         }
 
         return value;
-    };
+    }
+
+    // Name in the format of 'errorCatchNotifierWrap(MyComponentName)
+    // Note that this is only intended to show up in React PropType errors,
+    // stack traces won't read this value.
+    wrappedFunction.displayName = `errorCatchNotifierWrap(${targetFunction.name})`;
```
Comment (shaine) [inline src/error-catch-notifier.js:227]: Add a displayName property for React to read when showing JSX proptype errors. Should only apply to SFCs

File: test/overstock-error-catch-notifier-test.js [hunk 2]
```diff
@@ -428,16 +428,19 @@ describe('error-catch-notifier', function () {
...
             initErrorCatchNotifier(errorSubscribers, true, false);
 
-            wrap(errorFunction)();
+            const wrapped = wrap(errorFunction);
+            wrapped();
 
             expect(resultError)
                 .to.equal(testError);
+            expect(wrapped.displayName)
+                .to.equal('errorCatchNotifierWrap(testFunction)');
```
Comment (shaine) [inline test/overstock-error-catch-notifier-test.js:443]: This is the intended format of displayName

**Review by elliotec — APPROVED** [review]
cool

**Comment by shaine** [comment]
Tested the added behavior here: https://codesandbox.io/s/5421E5x

**Review by rianfowler — APPROVED** [review]

**Pull request merged**